src/main/scala/org/constellation/domain/redownload/RedownloadService.scala

```diff
@@ -330,16 +330,19 @@ class RedownloadService[F[_]: NonEmptyParallel](
     def wrappedRedownload(
       shouldRedownload: Boolean,
       meaningfulAcceptedSnapshots: SnapshotsAtHeight,
-      meaningfulMajorityState: SnapshotsAtHeight
+      meaningfulMajorityState: SnapshotsAtHeight,
+      maxMajorityHeight: Long
     ) =
       for {
         _ <- if (shouldRedownload) {
-          val plan = calculateRedownloadPlan(meaningfulAcceptedSnapshots, meaningfulMajorityState)
+          val redownloadPlan = calculateRedownloadPlan(meaningfulAcceptedSnapshots, meaningfulMajorityState)
+          val downloadPlan = calculateDownloadPlan(redownloadPlan, maxMajorityHeight)
+          val plan = if (isDownload) downloadPlan else redownloadPlan
```

Comment:
There is no need to calculate both `val` which are mutually exclusive.

I would just do

```scala
val plan =
  if (isDownload)
    calculateDownloadPlan(redownloadPlan, maxMajorityHeight)
  else
    calculateRedownloadPlan(meaningfulAcceptedSnapshots, meaningfulMajorityState)
```

Reply:
The download plan depends on redownload plan, that's why they're in sequence.

Reply:
looks like download plan uses redownload plan, but I'd just apply trim on the redownloadPlan conditionally

Reply:
Ah right, I missed that :D